go/actions/test/action.yml

```diff
@@ -0,0 +1,76 @@
+name: Test go extractor
+description: Run build, QL tests and optionally basic code sanity checks (formatting and generation)
+inputs:
+  go-version:
+    description: Which Go version to use for running the tests
+    required: false
+    default: ~1.22.0
+  run-code-checks:
+    description: Whether to run formatting, code and qhelp generation checks
+    required: false
+    default: false
+runs:
+  using: composite
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ inputs.go-version }}
+        cache: false
+      id: go
+
+    - name: Set up CodeQL CLI
+      uses: ./.github/actions/fetch-codeql
+
+    - name: Enable problem matchers in repository
+      shell: bash
+      run: 'find .github/problem-matchers -name \*.json -exec echo "::add-matcher::{}" \;'
+
+    - name: Build
+      shell: bash
+      run: |
+        bazel run go:create-extractor-pack
```

Comment:
It may be good to add a comment here noting that this explicitly does not run `make` (and therefore no `bazel run :gen`) because we want generated files to be checked-in. This uses whatever is checked-in for the extractor pack and the later check then complains if the generated files aren't up-to-date.

Related question: Might we want the "Check checked-in generated code" step to take place before "Build" to fail the build if the generated files are not up-to-date, rather than performing this step with whatever is checked-in?